WordPress/gutenberg
# Extract navigation link rendering code from the navigation block

## Description
Fixes #20178. Makes the Navigation Link block dynamic and relocates much of the rendering to this block.

For this PR, I've had to change how some features work:
- The submenu icons are now hidden using a `display: none;` style when turned off. Previously they weren't rendered at all.
- Font size and text color styles when viewing a post continue to work through CSS inheritance, the navigation link doesn't have any specific classnames or inline styles set to determine those colors.
- Background color no longer works when viewing a post, for the same reason as above, there are no specific classnames or inline styles set on the navigation link It's harder for the background colors to be inherited, as this isn't the natural behavior for most elements in CSS.

All of the above points could be resolved/improved by pushing forward the proposal in https://github.com/WordPress/gutenberg/issues/19685.

Server side rendered blocks could declare a context for child blocks, which would allow passing attributes to child Navigation Link blocks, and the rendering could work much as it did before. 

## How has this been tested?
1. Add a navigation block
2. Build out navigation block with lots of nested navigation links.
3. Preview the post.
4. Observe that the navigation still renders

## Screenshots 

## Types of changes
Code quality improvement





## Checklist:
